### PR TITLE
Refactor form submission

### PR DESCRIPTION
**Pull request by timbot1789**
Small refactor to form upload so cross pod uploads and self uploads hit the same code path + have unit tests.

Also fixes an issue with the notifications where they wouldn't display in the proper sequence

### CODE REVIEW

File: src/components/Form/helpers/FormSubmissionHelper.test.js
Comment (leekahung) [inline src/components/Form/helpers/FormSubmissionHelper.test.js:1]: I think we can move helper to utils or some other location. This way we could keep our components separate functionality.

File: src/components/Form/helpers/FormSubmissionHelper.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { uploadDocument, updateDocument, runNotification } from '../../../utils';
+
+const makeHandleFormSubmission =
```
Comment (leekahung) [inline src/components/Form/helpers/FormSubmissionHelper.js]: Add some basic JSDoc about the function is about and add basic types to the parameters

**Review by leekahung — CHANGES_REQUESTED** [review]
Made a few comments on the PR. See feedback.

File: src/components/Form/helpers/FormSubmissionHelper.js (hunk 1 above)
Comment (timbot1789) [reply src/components/Form/helpers/FormSubmissionHelper.js]: fixed

File: src/components/Form/helpers/FormSubmissionHelper.test.js
Comment (timbot1789) [reply src/components/Form/helpers/FormSubmissionHelper.test.js:1]: fixed

**Review by leekahung — APPROVED** [review]
Saw the revisions made. Looks good to me.

**Pull request merged**